### PR TITLE
feat: add discovery support

### PR DESCRIPTION
**Pull request by embbnux**

### CODE REVIEW

File: sdk/src/platform/Discovery.ts [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+            this._externalRetryMaxCount = initialData.retryCount;
+            this._externalRetryInterval = initialData.retryInterval;
+        }
+        this._initialized = true;
+        this.emit(events.initialized, initialData);
+    }
+
+    public async fetchInitialData() {
+        if (!this._initialFetchPromise) {
+            this._initialFetchPromise = this._fetchInitialData();
```
Comment (kirill-konshin) [inline sdk/src/platform/Discovery.ts:201]: Move this into `try {}` block.

File: sdk/src/platform/Discovery.ts [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+            );
+            const initialData = await response.json();
+            await this._setInitialData(initialData);
+            this._initialRetryCount = 0;
+            return initialData;
+        } catch (e) {
+            this._initialRetryCount += 1;
+            if (this._initialRetryCount < this._initialRetryMaxCount) {
+                await delay(this._initialRetryInterval * 1000);
+                return this._fetchInitialData();
```
Comment (kirill-konshin) [inline sdk/src/platform/Discovery.ts:233]: Do we want to retry on ALL errors? What if not network related errors will occur?

**Review by kirill-konshin — CHANGES_REQUESTED** [review]

File: sdk/src/platform/Discovery.ts (hunk 2 above)
Comment (embbnux) [reply sdk/src/platform/Discovery.ts:233]: Good point. we should not retry for 401

**Comment by kirill-konshin** [comment]
CI needs to be fixed before merge.

**Comment by embbnux** [comment]
@kirill-konshin updated to add tests and fix CI

**Pull request merged**